File: lib/src/main/java/uk/iatom/rfm69/registers/Preamble.java
package uk.iatom.rfm69.registers;

public enum Preamble {
    

    // RegPreamble),
    SIZE_MSB_VALUE(0x00), // Default),
    SIZE_LSB_VALUE(0x03); // Default),;

    private byte val;
    
    Preamble(int val) {
        this.val = (byte) val;
    }

    public byte getVal() {
        return val;
    }
}

File: lib/src/main/java/uk/iatom/rfm69/register/values/Preamble.kt
package uk.iatom.rfm69.register.values

import uk.iatom.rfm69.register.IRegisterValue


class Preamble {


    enum class Msb(`val`: Int): IRegisterValue { // RegPreamble),
        SIZE(0x00);

        private val `val`: Byte

        init {
            this.`val` = `val`.toByte()
        }

        override fun `val`(): Byte {
            return `val`
        }
    }

    enum class Lsb(`val`: Int): IRegisterValue { // RegPreamble),

        SIZE(0x03);

        private val `val`: Byte

        init {
            this.`val` = `val`.toByte()
        }

        override fun `val`(): Byte {
            return `val`
        }
    }
}


